modules/c++/sys/source/OSUnix.cpp

```diff
@@ -189,9 +189,14 @@ std::string sys::OSUnix::getTempName(const std::string& path,
     else
     {
         name = tempname;
+        sys::File newFile(name, sys::File::READ_AND_WRITE, sys::File::CREATE);
         free(tempname);
```

Comment:
I'm not sure I can get rid of the buffer completely. mkstemp needs a char * that it can modify, and string.c_str() is const. I think the best I can do is to build the it up with std::string, then copy into the buffer before I pass it.